src/core/metrics.cc

```diff
@@ -211,8 +211,8 @@ metric_value metric_value::operator+(const metric_value& c) {
     return res;
 }
 
-void metric_value::ulong_conversion_error(double d) {
-    throw std::range_error(format("cannot convert double value {} to unsigned long", d));
+void metric_value::ulong_conversion_error(double d, const char* ctype) {
+    seastar_logger.warn("cannot convert double value {} to {}", d, ctype);
```

Comment:
Please add some context to the messege - who can't convert, why it was converted?
Aren't we risking log spam here? When does this warning appear - just when querying the metrics? When updating the metrics? Or what?

Reply:
> Please add some context to the messege

to which message?  the git commit message?

>  - who can't convert, why it was converted?

I don't know

> Aren't we risking log spam here?

The risk is the same as before this patch since there was a corresponding error about the unhandled exception.

> When does this warning appear - just when querying the metrics? When updating the metrics? Or what?

It's really hard to tell since the backtrace we got from the unhandled exception starts at an unidentified background seastar task.
Based on https://github.com/scylladb/scylladb/issues/9793#issuecomment-997370342, it seems to come from `scollectd::cpwriter::write_value`

Reply:
> > Please add some context to the messege
> 
> to which message? the git commit message?

No, the log message. I imagine a user seeing "cannot convert double value {} to {}" a thousand times in the log and having absolutely no idea what this relates to - you aren't even saying that this is related to metrics, let alone which metric this is.
 
> > When does this warning appear - just when querying the metrics? When updating the metrics? Or what?
> 
> It's really hard to tell since the backtrace we got from the unhandled exception starts at an unidentified background seastar task. Based on [scylladb/scylladb#9793 (comment)](https://github.com/scylladb/scylladb/issues/9793#issuecomment-997370342), it seems to come from `scollectd::cpwriter::write_value`